samcli/commands/local/lib/local_lambda.py

```diff
@@ -75,10 +75,13 @@ def invoke(self, function_name, event, stdout=None, stderr=None):
         """
 
         # Generate the correct configuration based on given inputs
-        function = self.provider.get(function_name)
+        function = None
+        for f in self.provider.get_all():
+            if f.functionname == function_name:
+                function = f
```

Comment:
Not sure I understand the move from `.get(function_name)` to `.get_all()`. We should still be able to fetch the function based on the name here right? I think the decision of logical id or function name can be determined directly in the `.get(function_name)` call on the provider.

Reply:
I was worried about changing too much but I think you're approach is correct